### ISSUE TITLE
WCAG 2.1 accessibility requirement for color contrast

### ISSUE THREAD

**Issue by lukecanvin** [salesforce-ux/design-system#563]
I greatly value the efforts from the team to make SLDS highly accessible. Your accessibility documentation states that you follow the "two main rules of accessibility as it relates to color", and these come from the WCAG 2.0 specification ([1.4.1](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html) and [1.4.3](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html)).

The WCAG 2.1 specification was released in June and it contains a new color contrast requirement:

> **1.4.11 Non-text Contrast**
> The visual presentation of the following have a contrast ratio of at least 3:1 against adjacent color(s):
> 
> **User Interface Components**
> Visual information required to identify user interface components and states, except for inactive components or where the appearance of the component is determined by the user agent and not modified by the author;
>
> **Graphical Objects**
> Parts of graphics required to understand the content, except when a particular presentation of graphics is essential to the information being conveyed.

All the details can be found on [Understanding Success Criterion 1.4.11: Non-text Contrast](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html)

This is a AA criterion and so will need to be widely supported; by Salesforce too I expect.

Will you be reviewing the SLDS tokens/components in light of this? I can see quite a few places where changes would need to be made, some basic examples being input fields, neutral buttons, scoped/vertical tabs, etc. Such a change would probably necessitate some careful thought about the new colors to use and how to apply them to existing tokens.

**Comment by SiTaggart** [comment]
We have been following the development of WCAG 2.1 very closely and welcome the additions and improvements to the standard. Beyond colour contrast, the new version introduces a number of quite large changes which will have a significant impact on our designs and design system.

We are in the very early stages of planning how to weave these updates into the system and our products. It'll take some time to do it properly. Thanks for your interest

**Comment by SiTaggart** [comment]
I'm going to go ahead and close this for now. We know it's an issue, we are more than likely going to accommodate the changes from WCAG 2.1 into the system eventually but it'll take a while as we'll need to address things bit by bit.

Thanks for your comment and appreciation of our accessibility efforts.

**Comment by Dan-Wuensch** [comment]
@SiTaggart and team - what's the latest on this issue? Our company is running into accessibility audit failures due to use of SLDS as pertains to color contrast requirements on the newest version. Is there a timeline for addressing?
